bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingConfigDescriptionProvider.java

```diff
@@ -0,0 +1,91 @@
+package org.eclipse.smarthome.core.thing.internal;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Locale;
+
+import org.eclipse.smarthome.config.core.ConfigDescription;
+import org.eclipse.smarthome.config.core.ConfigDescriptionProvider;
+import org.eclipse.smarthome.config.core.ConfigDescriptionRegistry;
+import org.eclipse.smarthome.config.core.ConfigOptionProvider;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.type.ThingType;
+import org.eclipse.smarthome.core.thing.type.ThingTypeRegistry;
+
+/**
+ * Provides a proxy for thing configuration descriptions.
+ *
+ * If a thing config description is requested, the provider will look up the thingType
+ * to get the configURI, get the config description for the thingType. If the thingHandler
+ * supports the {@link ConfigOptionProvider} interface, it will call the getParameterOptions
+ * method to get updated options.
+ *
+ * @author Chris Jackson - Initial Implementation
+ *
+ */
+public class ThingConfigDescriptionProvider implements ConfigDescriptionProvider {
+    private ThingRegistry thingRegistry;
+    private ThingTypeRegistry thingTypeRegistry;
+    private ConfigDescriptionRegistry configDescriptionRegistry;
+
+    protected void setThingRegistry(ThingRegistry thingRegistry) {
+        this.thingRegistry = thingRegistry;
+    }
+
+    protected void unsetThingRegistry(ThingRegistry thingRegistry) {
+        this.thingRegistry = null;
+    }
+
+    protected void setConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = configDescriptionRegistry;
+    }
+
+    protected void unsetConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = null;
+    }
+
+    protected void setThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
+        this.thingTypeRegistry = thingTypeRegistry;
+    }
+
+    protected void unsetThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
+        this.thingTypeRegistry = null;
+    }
+
+    @Override
+    public Collection<ConfigDescription> getConfigDescriptions(Locale locale) {
+        return null;
```

Comment:
this means that /config-descriptions won't show anything for things. This is not necessarily expected, but probably the best way to deal with it - returning a description for every single thing is probably a bit too much.
@dnobel Is this ok for you or could that have other side effects?